### DISCUSSION TITLE
Ability to use "slots/lists" instead of permutation

### DISCUSSION THREAD

**Discussion by hybridherbst** [hlky/stable-diffusion#177] [Ideas]
I find permutations really useful, but right now it's a bit hard to do things like

`Shapes on a background |  The background is red | The background is green | The shapes are blue | The shapes are white`

since of course there's many permutations that don't make sense.

Would be great to have a syntax to define slots, e.g.

```
<slot1> Shapes on a <slot2> background 
slot1: blue | white
slot2: red | green
```

Which would generate more reasonable (and less) combinatorial variants.

Thanks!

Example of "useless" permutations:
![image](https://user-images.githubusercontent.com/2693840/187725314-8ab574c4-be86-4617-9ed8-f90aae302699.png)

**Comment by lozenge** [comment]
Assuming I have understood you correctly, this is actually already an existing feature! Credit goes to @JustOxlamon, who implemented it a few days ago. It's [mentioned in the README](https://github.com/hlky/stable-diffusion-webui#prompt-combinations). Easy to miss, but very powerful to use. 

For your example, you could do `@(blue|white) shapes on a (red|green) background`, which results in 4 prompts. You can add another vertical bar to the end of each if you wanted to also test the absence of each group, resulting in 9 prompts.

**Reply by hybridherbst** [reply]
Nice, I must have indeed missed that!

**Reply by hybridherbst** [reply]
Might be worth noting in the UI that the "Advanced > Prompt Matrix" needs to be on for this to work - I guess it might need to be renamed to "Advanced > Enable Prompt Syntax" or something like that.
